compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3747,6 +3784,30 @@ private boolean isAssignableToUnionType(BType source, BType target, Set<TypePair
         return true;
     }
 
+    private boolean isUnionDistinctError(Set<BType> memberTypesSet) {
```

Comment:
Shall we rename this method? We are not passing an error to this but a set of type ids. So this is doing some uniqueness check on a type id set right?

Reply:
Changed